boyscout & consolidate menu items

fabiangotzen opened the pull request:
Most of it is just fixing TSLint errors, I've also tried to simplify the menu: 

![image](https://user-images.githubusercontent.com/7059532/110773056-9bf41980-825c-11eb-9755-dca0ee1f9309.png)

So you directly navigate to the list component. From there you have the posibility to add the flight/glider/place: 
![image](https://user-images.githubusercontent.com/7059532/110773173-bfb75f80-825c-11eb-8750-9057ce09561f.png)
![image](https://user-images.githubusercontent.com/7059532/110773847-774c7180-825d-11eb-9894-87f92741ebb6.png)
![image](https://user-images.githubusercontent.com/7059532/110773975-977c3080-825d-11eb-8a41-cd4f499d15f8.png)

laggery commented on src/app/account/register/register.page.ts:
```diff
@@ -47,7 +48,7 @@ export class RegisterPage implements OnInit, OnDestroy {
         (async (error: any) => {
           await loading.dismiss();
-          if (error.status === 409) {
+          if (error.status === 4) {
```
What is the error code 4?

fabiangotzen replied:
typo, fixed:  HttpStatusCode.CONFLICT

laggery commented on src/app/glider/glider-list/glider-list.page.ts:
```diff
@@ -35,25 +43,27 @@ export class GliderListPage implements OnInit, OnDestroy, AfterViewInit {
+      .subscribe(async (res: Glider[]) => {
+        // @hack for hide export item
+        setTimeout(async () => {
+          this.content.scrollToPoint(0, 48);
```
Maybe also add await like in flight-list?

fabiangotzen replied:
done

laggery commented on src/app/place/place-list/place-list.page.ts:
```diff
@@ -36,11 +36,13 @@ export class PlaceListPage implements OnInit, OnDestroy, AfterViewInit {
+      .subscribe(async (res: Place[]) => {
       // @hack for hide export item
       setTimeout(async () => {
         this.content.scrollToPoint(0, 48);
```
Maybe also add await like in flight-list?

fabiangotzen replied:
done

laggery reviewed (commented):
Design bug:
![image](https://user-images.githubusercontent.com/2563295/110783379-aa483280-8268-11eb-90c9-cbe428ae11f5.png)

The "add" button is at the very end of the list. To do this I first have to load 1000 flights
Same for place list and glider list

fabiangotzen commented:
> Design bug:
> ![image](https://user-images.githubusercontent.com/2563295/110783379-aa483280-8268-11eb-90c9-cbe428ae11f5.png)
> 
> The "add" button is at the very end of the list. To do this I first have to load 1000 flights
> Same for place list and glider list

I took some inspiration from the android add contacts list: (moved filter there as well)

![image](https://user-images.githubusercontent.com/7059532/110807253-22235680-8283-11eb-90da-5eb69d5dcbe2.png)

let me know what you think.

fabiangotzen commented:
Also, a small redesign of the statistics page: 
![image](https://user-images.githubusercontent.com/7059532/110807429-53038b80-8283-11eb-8d35-dc8492196637.png)
(increased font size + put them into ion-cards)

laggery commented:
> Also, a small redesign of the statistics page:
> ![image](https://user-images.githubusercontent.com/7059532/110807429-53038b80-8283-11eb-8d35-dc8492196637.png)
> (increased font size + put them into ion-cards)

Design:
- The idea to have bigger font is good but when you have a lot of flights it cause some design problem
- For me, the filter is not on the right place because it also filter the graphics
- Card on statistics make the graphics lot smaller on mobile and I find we lose to much places
![image](https://user-images.githubusercontent.com/2563295/110841627-6d9b2c00-82a6-11eb-927a-2ee1959cdb86.png)

About "add" and "Filter" button.
Filter button -> in my opinion not clear enough that it is the filter button. It is also not anytime accessible. For example when i scroll down in my flights and then I want to change the filter I need first to scroll up
The add button is small but takes a lot of place so. I don't have a better solution at the moment.

fabiangotzen commented:
> Design:
> 
> * The idea to have bigger font is good but when you have a lot of flights it cause some design problem
> * For me, the filter is not on the right place because it also filter the graphics
> * Card on statistics make the graphics lot smaller on mobile and I find we lose to much places
>   ![image](https://user-images.githubusercontent.com/2563295/110841627-6d9b2c00-82a6-11eb-927a-2ee1959cdb86.png)
> 
> About "add" and "Filter" button.
> Filter button -> in my opinion not clear enough that it is the filter button. It is also not anytime accessible. For example when i scroll down in my flights and then I want to change the filter I need first to scroll up
> The add button is small but takes a lot of place so. I don't have a better solution at the moment.

* Adjusted the statistics page: 
![image](https://user-images.githubusercontent.com/7059532/110925469-8564c580-8323-11eb-86b4-e0e23f35e37a.png)
*reverted changes to filter and added the "add" buttons to the footer which is always there.
![image](https://user-images.githubusercontent.com/7059532/110925628-bba24500-8323-11eb-9285-d5585d9fd87d.png)

fabiangotzen commented:
![image](https://user-images.githubusercontent.com/7059532/110945402-edc0a080-833d-11eb-8921-3956896ae4a9.png)
moved it to the top
